src/coreclr/tools/Common/Compiler/InstructionSetSupport.cs

```diff
@@ -333,12 +334,43 @@ public bool ComputeInstructionSetFlags(int maxVectorTBitWidth,
             {
                 unsupportedInstructionSets.AddInstructionSet(instructionSetConversion[unsupported]);
             }
+
             unsupportedInstructionSets.ExpandInstructionSetByReverseImplication(_architecture);
             unsupportedInstructionSets.Set64BitInstructionSetVariants(_architecture);
 
             if ((_architecture == TargetArchitecture.X86) || (_architecture == TargetArchitecture.ARM))
                 unsupportedInstructionSets.Set64BitInstructionSetVariantsUnconditionally(_architecture);
 
+            // While it's possible to enable individual AVX-512 ISA's, it is not
```

Comment:
We still need to adjust the assert that raised this issue, right? Because it's happening before `ComputeInstructionSetFlags` is called? -- I'd presume that also entails moving the "opportunistic" additions to after we've expanded the API set here?

Reply:
As is, I think the assert that triggered this issue is fine. My reasoning for this is that it's running on `supportedInstructionSet`:

https://github.com/dotnet/runtime/blob/086ee6680a28eb2384a42ed350a536c32e07ad6d/src/coreclr/tools/Common/InstructionSetHelpers.cs#L214-L216

This variable contains the result of running the expansions on the original instruction set builder above, on line ~170, which means there's nothing else to do to it after the assert:

https://github.com/dotnet/runtime/blob/086ee6680a28eb2384a42ed350a536c32e07ad6d/src/coreclr/tools/Common/InstructionSetHelpers.cs#L170-L172

The one that is getting expanded after the assert is the `optimisticInstructionSetSupportBuilder`, which isn't causing issues. Now however, this does raise the question. Why are we not asserting on the optimistic one too? As in there is no assert code for it. Should we move the optimistic computations earlier and assert both of them at once at the end? Should we leave it as is? What are your thoughts Tanner?

Reply:
Ah, I see, there are two `ComputeInstructionSetFlags` calls, so the assert should be correct.

Reply:
> Now however, this does raise the question. Why are we not asserting on the optimistic one too? As in there is no assert code for it. Should we move the optimistic computations earlier and assert both of them at once at the end? Should we leave it as is? What are your thoughts Tanner?

There's a general expectation that `opportunistic` is a superset of `supported`, so I'm not sure we need to repeat the assertion.

I don't think it would hurt to do so, however.

Reply:
Okok! Sounds good. I will then add the asserts for the opportunistic ones too and wait for the pipelines to come back green before merging. Thanks a lot for your guidance on this Tanner!